app/scripts/modules/core/config/settings.ts

```diff
@@ -0,0 +1,174 @@
+    region: string;
+  };
+}
+
+interface IAWSProviderSettings {
+  defaults: {
+    account: string;
+    region: string;
+    iamRole: string;
+    subnetType?: string;
+  };
+  defaultSecurityGroups: string[];
+  loadBalancers: {
+    inferInternalFlagFromSubnet: boolean;
+    certificateTypes?: string[];
+  };
+  useAmiBlockDeviceMappings: boolean;
+  classicLaunchLockout?: number;
+  metrics?: {
+    customNamespaces?: string[];
+  };
+}
+
+interface IGCEProviderSettings {
+  defaults: {
+    account: string;
+    region: string;
+    zone: string;
+  };
+}
+
+interface ITitusProviderSettings {
+  defaults: {
+    account: string;
+    region: string;
+    iamProfile: string;
+  };
+}
+
+interface IOpenStackProviderSettings {
+  defaults: {
+    account: string;
+    region: string;
+  };
+}
+
+interface IKubernetesProviderSettings {
+  defaults: {
+    account: string;
+    namespace: string;
+    proxy: string;
+  };
+}
+
+interface IAppengineProviderSettings {
+  defaults: {
+    account: string;
+    editLoadBalancerStageEnabled: boolean;
+  };
+}
+
+
+interface IProviderSettings {
+  azure?: IAzureProviderSettings;
+  aws?: IAWSProviderSettings;
+  gce?: IGCEProviderSettings;
+  titus?: ITitusProviderSettings;
+  openstack?: IOpenStackProviderSettings;
+  kubernetes?: IKubernetesProviderSettings;
+  appengine?: IAppengineProviderSettings;
+  [key: string]: any; // allows custom providers not typed in here (good for testing too)
+}
+
+interface INotificationSettings {
+  email: {
+    enabled: boolean;
+  };
+  hipchat: {
+    enabled: boolean;
+    botName: string;
+  };
+  sms: {
+    enabled: boolean;
+  };
+  slack: {
+    enabled: boolean;
+    botName: string;
+  };
+}
+
+export interface ISpinnakerSettings {
+  checkForUpdates: boolean;
+  debugEnabled: boolean;
+  defaultProviders: string[];
+  feedbackUrl: string;
+  gateUrl: string;
+  bakeryDetailUrl: string;
+  authEndpoint: string;
+  pollSchedule: number;
+  defaultTimeZone: string; // see http://momentjs.com/timezone/docs/#/data-utilities/
+  defaultCategory: string;
+  defaultInstancePort: number;
+  providers?: IProviderSettings;
+  whatsNew: {
+    gistId: string;
+    fileName: string;
+    accessToken?: string;
+  };
+  notifications: INotificationSettings;
+  authEnabled: boolean;
+  authTtl: number;
+  gitSources: string[];
+  triggerTypes: string[];
+  analytics: {
+    ga?: boolean;
+  };
+  feature: {
+    entityTags?: boolean;
+    fiatEnabled?: boolean;
+    pipelines?: boolean;
+    notifications?: boolean;
+    fastProperty?: boolean;
+    vpcMigrator?: boolean;
+    clusterDiff?: boolean;
+    roscoMode?: boolean;
+    netflixMode?: boolean;
+    chaosMonkey?: boolean;
+    // whether stages affecting infrastructure (like "Create Load Balancer") should be enabled or not
+    infrastructureStages?: boolean;
+    jobs?: boolean;
+    snapshots?: boolean;
+    dockerBake?: boolean;
+    blesk?: boolean;
+    tableau?: boolean;
+    [key: string]: any;
+  };
```

Comment:
It can. But having all the ones we know about provides typeahead in IDEs, which is nice. If you don't think that's worth it, I can take it out.